src/sidebar/util/term-before-position.ts

```diff
@@ -28,16 +26,20 @@ export function getContainingWordOffsets(
   text: string,
   referencePosition: number,
 ): WordOffsets {
-  const precedingEmptyCharPos = text
-    .slice(0, referencePosition)
-    .search(/\s\S*$/);
-  const subsequentEmptyCharPos = text.slice(referencePosition).search(/\s/);
+  const precedingText = text.slice(0, referencePosition);
+  const matches = [...precedingText.matchAll(/[\s,.;:|?!'"\-()[\]{}]/g)];
+  const precedingCharPos =
+    matches.length > 0 ? Math.max(...matches.map(match => match.index)) : -1;
```

Comment:
This looks for the position of the last occurrence of any of the empty chars or punctuation chars, in the text before the reference position.